tests/assertions/is_substring_of.zsh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env zunit
+
```

Comment:
do we want the empty string to match for `is_substring_of`? This case fails in shell script

`[[ *''* = "some string" ]]`

so in order to be consistent I guess we should also follow that pattern?